docs/pages/access-controls/idps/saml-attribute-mapping.mdx

```diff
@@ -0,0 +1,194 @@
+
+You can also add an entirely new value that is not available in the user spec file.
+E.g.:
+Expression | Result
+-----------| ------
+`set().add("prod-ssh")` or `set("prod-ssh")` | `prod-ssh`
+
+#### `remove`
+Remove value(s). Works on `user.spec.roles` and `user.spec.traits.groups`
+Expression | Result
+---------- | ------
+`user.spec.roles.remove("editor", "access")` | `dev-ssh`
+
+#### `contains`
+Returns boolean value for matching expression. To be used in helper functions such as `ifelse`.
+Works on `user.spec.roles` and `user.spec.traits.groups`.
+Expression | Result
+---------- | ------
+`user.spec.traits.groups.contains("okta-admin")` | `true`
+
+### Helper functions
+#### `strings.upper`
+Transform string to upper.
+Expression | Result
+-----------| ------
+`strings.upper(user.spec.traits.firstname)` | `FOO`
+
+#### `strings.lower`
+Transform string to lower.
+Expression | Result
+-----------| ------
+`strings.lower(user.spec.traits.lastname)` | `bar`
+
+#### `strings.replaceall`
+Replace all matching strings.
+Expression | Result
+-----------| ------
+`strings.replaceall(user.spec.traits.groups, "-", "+")` | `okta+admin, dev+sso, dev+rdp`
+`strings.replaceall(user.spec.traits.groups, "admin", "dev")` | `okta-dev, dev-sso, dev-rdp`
+
+#### `strings.split`
+Split string at matching character.
+Expression | Result
+-----------| ------
+`strings.split(user.spec.traits.groups, "-")` | `okta, admin, dev, sso, rdp`
+
+#### `ifelse`
+Conditionally return values. To be used in conjunction with methods such as `contains`.
+
+Signature: `ifelse(condition, "value to return if condition is true", "value to return if condition is false")`
+Expression | Result
+-----------| ------
+`ifelse(user.spec.traits.groups.contains("okta-admin"), user.spec.traits.groups.add("newgroup"), user.spec.traits.groups)` | `okta-admin, dev-sso, dev-rdp, newgroup`
+
+#### `union`
+Combine values in `user.spec.traits.groups` and `user.spec.roles`.
+Expression | Result
+-----------| ------
+`union(user.spec.traits.groups, user.spec.roles)` | `okta-admin, dev-sso, dev-rdp, access, editor, dev-ssh`
+`union(user.spec.traits.groups.remove("okta-admin"), user.spec.roles)` | `dev-sso, dev-rdp, access, editor, dev-ssh`
+
+
+## Testing attribute mapping
+### Prerequisites
+
+(!docs/pages/includes/no-oss-prereqs-tabs.mdx!)
+
+- (!docs/pages/includes/tctl.mdx!)
+- User with permission to create Service Provider resource. The preset `editor` role has this permission.
+
+### `test-attribute-mapping` command
+
+Attribute mapping can be tested using `tctl idp saml test-attribute-mapping` command.
+`test-attribute-mapping` command accepts three arguments.
+
+- `--users`: user names or names of files containing user spec. Required.
+- `--sp`: name of file containing service provider spec with attribute mapping. Required.
+- `--format`: `yaml` or `json`.  Optional. Text output by default if the flag is not provided.
```

Comment:
So it supports `text` output format too then?

Reply:
It does if `--format` is not provided.
Now I think, I should have also added `text` in the `--format` : `--format: yaml | text | json` to avoid surprises.